File: Java/src/main/java/QuestionBank/Solution501_600/Solution590.java
package QuestionBank.Solution501_600;

import tool.Node;

import java.util.ArrayList;
import java.util.List;

/**
 * @author windows
 */
public class Solution590 {

    ArrayList<Integer> list = new ArrayList<>();
    public List<Integer> postorder(Node root) {
        if (root == null){
            return list;
        }
        List<Node> children = root.children;
        if (children != null){
            for (int i = 0; i < children.size(); i++) {
               postorder(children.get(i));
            }
        }
        list.add(root.val);
        return list;
    }


    public static void main(String[] args) {
        Solution590 solution590 = new Solution590();
    }
}

File: C++/src/Solution501_600/Solution590.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution590.h"
#include <functional>
#include <iostream>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;


// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

Node() {}

Node(int _val) {
  val = _val;
}

Node(int _val, vector<Node*> _children) {
  val = _val;
  children = _children;
}
};

vector<int> postorder(Node* root) {
  vector<int> res;
  function<void(Node*)> dfs = [&](Node* root) {
    if (!root) return;
    for (auto child : root->children) {
      dfs(child);
    }
    res.push_back(root->val);
  };
  dfs(root);
  return res;
}